spec/spans.yml

```
@@ -0,0 +1,174 @@
+groups:
+  - id: span.graphql.server.operation
+    type: span
+    stability: development
+    span_kind: server
+    brief: >
+      This span represents an incoming GraphQL operation on a server implementation.
+    note: |
+      **Span name** SHOULD be of the format `{graphql.operation.type}` provided
+      `graphql.operation.type` is available. If `graphql.operation.type` is not available,
+      the span SHOULD be named `GraphQL Operation`.
+
+      For persisted operations with a specified operation name, instrumentations MAY provide
+      a configuration option to enable a more descriptive span name following
+      `{graphql.operation.type} {graphql.operation.name}` format when
+      `graphql.operation.name` is available and the operation is successfully identified
+      in the document.
+
+      > **Warning**
+      > The `graphql.operation.name` value is provided by the client and can have high
+      > cardinality. Using it in the GraphQL server span name is NOT RECOMMENDED for
+      > ad-hoc operations without careful consideration of cardinality implications.
+      > For persisted operations, the cardinality is bounded and using the operation
+      > name in the span name is more acceptable.
```

```suggestion
      For operation domains with bounded cardinality, instrumentations MAY provide
      a configuration option to enable a more descriptive span name following the
      `{graphql.operation.type} {graphql.operation.name}` format when
      `graphql.operation.name` is available and the operation is successfully identified
      in the document.

      > **Warning**
      > The `graphql.operation.name` value is provided by the client and can have high
      > cardinality. Using it in the GraphQL server span name is NOT RECOMMENDED for
      > ad-hoc operations without careful consideration of cardinality implications.
      > For persisted operations or internal GraphQL APIs, the cardinality is bounded and using the operation
      > name in the span name is more acceptable.
```

I think it's clearer to focus on the condition causing the risk (cardinality), and use "persisted operations" only only exemplify, instead of leading with "persisted operations", as one could interpret "persisted operations" as being the main deciding factor that applies here.